consent/api/v1/views.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+"""
+A generic API for edX Enterprise's Consent application.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+from consent.api import permissions
+from consent.errors import ConsentAPIRequestError
+from consent.helpers import (
+    consent_exists,
+    consent_provided,
+    consent_required,
+    get_enterprise_customer,
+    get_enterprise_customer_user,
+    get_user_id,
+)
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_400_BAD_REQUEST, HTTP_404_NOT_FOUND
+from rest_framework.views import APIView
+
+# Will be using internal models at a later time.
+from enterprise.models import EnterpriseCourseEnrollment
+
+
+class DataSharingConsentView(APIView):
+    """
+        **Use Cases**
+
+            Presents a generic data sharing consent API to applications
+            that have Enterprise customers who require data sharing
+            consent from users.
+
+        **Behavior**
+
+            Implements GET, POST, and DELETE which each have roughly
+            the following behavior (see their individual handlers for
+            more documentation):
+
+            GET /consent/api/v1/data_sharing_consent?username=bob&enterprise_customer_uuid=ENTERPRISE-UUID&course_id=ID
+            >>> {
+            >>>     "username": "bob",
+            >>>     "course_id": "course-v1:edX+DemoX+Demo_Course",
+            >>>     "enterprise_customer_uuid": "enterprise-uuid-goes-right-here",
+            >>>     "exists": False,
+            >>>     "consent_provided": False,
+            >>>     "consent_required": True,
+            >>> }
+
+            If the "exists" key is false, then the body will be returned
+            with a 404 Not Found error code; otherwise, 200 OK. If either
+            of enterprise_customer_uuid or username is not provided, an
+            appropriate 400-series error will be returned.
+
+            POST or DELETE /consent/api/v1/data_sharing_consent
+            >>> {
+            >>>     "username": "bob",
+            >>>     "course_id": "course-v1:edX+DemoX+Demo_Course",
+            >>>     "enterprise_customer_uuid": "enterprise-uuid-goes-right-here"
+            >>> }
+
+            The API accepts JSON objects with these key-value pairs for
+            POST or DELETE.
+
+        **Notes**
+
+            `course_id` specifies a course key (course-v1:edX+DemoX),
+            and not a course run key (course-v1:edX+DemoX+Demo_Course).
+
+    """
+
+    permission_classes = [permissions.IsStaff, ]
+
+    REQUIRED_PARAM_USERNAME = 'username'
+    REQUIRED_PARAM_COURSE_ID = 'course_id'
+    REQUIRED_PARAM_ENTERPRISE_CUSTOMER = 'enterprise_customer_uuid'  # pylint: disable=invalid-name
+
+    CONSENT_EXISTS = 'exists'
+    CONSENT_GRANTED = 'consent_provided'
+    CONSENT_REQUIRED = 'consent_required'
+
+    def get_required_query_params(self, request):
```

Comment:
Ohhh. I must have been getting an error-code response, but the overall response was still being rendered fully. Maybe we should return an actual structurally-different response if the parameters are missing, rather than proceeding to do the database queries with the empty values but just flagging with a 400 code.